packages/fuel-indexer-schema/src/db/queries.rs

```diff
@@ -0,0 +1,305 @@
+use super::arguments::Filter;
+use fuel_indexer_database::DbType;
+use std::collections::{HashMap, HashSet};
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum QueryElement {
+    Field { key: String, value: String },
+    ObjectOpeningBoundary { key: String },
+    ObjectClosingBoundary,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct EntityFilter {
+    pub fully_qualified_table_name: String,
+    pub filters: Vec<Filter>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct JoinCondition {
+    pub referencing_key_table: String,
+    pub referencing_key_col: String,
+    pub primary_key_table: String,
+    pub primary_key_col: String,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct QueryJoinNode {
+    pub dependencies: HashMap<String, JoinCondition>,
+    pub dependents: HashMap<String, JoinCondition>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct UserQuery {
+    pub elements: Vec<QueryElement>,
+    pub joins: HashMap<String, QueryJoinNode>,
+    pub namespace_identifier: String,
+    pub entity_name: String,
+    pub filters: Vec<EntityFilter>,
+}
+
+impl UserQuery {
+    pub fn to_sql(&mut self, db_type: &DbType) -> String {
```

Comment:
When you add rust-docs it'd be nice to know what the overall strategy is here for this `to_sql` method. The more info the better.